streams/src/main/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdater.java

```diff
@@ -748,7 +748,7 @@ public Set<Task> getTasks() {
     @Override
     public boolean restoresActiveTasks() {
         return !executeWithQueuesLocked(
-            () -> getStreamOfNonPausedTasks().filter(Task::isActive).collect(Collectors.toSet())
+            () -> getStreamOfTasks().filter(Task::isActive).collect(Collectors.toSet())
```

Comment:
This is to make the case with state-updater consistent with the one without state-updater: when tasks are paused, they are still considered as the ones that need to be restoration completed, before we can transit the main thread to RUNNING.

See below the TODO marker on the interface API for details.